crates/extensions/aws_mapper_ext/src/converter.rs

```diff
@@ -42,60 +42,75 @@ impl AwsConverter {
+        channel: Channel,
         input: &MqttMessage,
     ) -> Result<Vec<MqttMessage>, ConversionError> {
-        // serialize with ThinEdgeJson for health, just add the timestamp
-        if input.topic.name.starts_with("tedge/health") {
-            let payload = self.with_time_stamp(input)?;
-            let topic_suffix = match input.topic.name.split_once('/') {
-                Some((_, topic_suffix)) => topic_suffix,
-                None => return Ok(vec![]),
-            };
-            let out_topic = Topic::new(&format!("aws/td/{topic_suffix}"))?;
-
-            let output = MqttMessage::new(&out_topic, payload);
-            self.size_threshold.validate(&output)?;
-            Ok(vec![(output)])
-        } else {
-            Ok(vec![])
+        // don't convert mosquitto bridge notification topic
+        // https://github.com/thin-edge/thin-edge.io/issues/2236
+        if source.as_str() == MOSQUITTO_BRIDGE_TOPIC_ID {
+            return Ok(vec![]);
+        }
+
+        match channel {
+            Channel::Measurement {
+                measurement_type: type_name,
+            }
+            | Channel::Event {
+                event_type: type_name,
+            }
+            | Channel::Alarm {
+                alarm_type: type_name,
+            } => self.convert_telemetry_message(input, source, &type_name),
+
+            Channel::Health => self.convert_health_message(&source, &channel, input),
+
+            _ => Ok(vec![]),
         }
     }
 
-    fn try_convert_te_topics(
-        &mut self,
-        source: EntityTopicId,
-        channel: Channel,
+    fn convert_health_message(
+        &self,
+        source: &EntityTopicId,
+        channel: &Channel,
         input: &MqttMessage,
     ) -> Result<Vec<MqttMessage>, ConversionError> {
-        let mea_type = match &channel {
-            Channel::Measurement { measurement_type } => measurement_type,
-            Channel::Event { event_type } => event_type,
-            Channel::Alarm { alarm_type } => alarm_type,
-            _ => return Ok(vec![]),
-        };
-        self.convert_message(input, source, mea_type)
+        match channel {
+            Channel::Health => (),
+            _ => panic!("Not a health message"),
+        }
+
+        let source = normalize_name(source);
+        let out_topic = format!("aws/td/{source}/status/health");
+        let payload = self.with_time_stamp(input)?;
+        let output = MqttMessage::new(&Topic::new(&out_topic).unwrap(), payload);
+
+        Ok(vec![output])
     }
 
-    fn convert_message(
+    fn convert_telemetry_message(
         &mut self,
         input: &MqttMessage,
         source: EntityTopicId,
         telemetry_type: &String,
     ) -> Result<Vec<MqttMessage>, ConversionError> {
         let payload = self.with_time_stamp(input)?;
-        let source = normalize_name(source);
+        let source = normalize_name(&source);
+        // XXX: should match on `Channel` instead
```

Comment:
Although the channel check is already done by the caller (`try_convert_te_topics()`), we need again to check channel doesn't seem efficient.

Reply:
I think what he meant is that, it's much better to get that channel passed here via an argument and match on the `channel` object than doing the array matches as done below. That exact same information is available in the channel, making the split/collect/match unnecessary/redundant.